services/QuillLMS/spec/workers/pdfs/admin_usage_snapshot_email_worker_spec.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+module Pdfs
+  RSpec.describe AdminUsageSnapshotEmailWorker, type: :worker do
+    subject { described_class.new.perform(pdf_subscription_id) }
+
+    let(:admin_report_filter_selection) { create(:admin_report_filter_selection, :with_default_filters) }
+    let(:pdf_subscription) { create(:pdf_subscription, admin_report_filter_selection:) }
+    let(:pdf_subscription_id) { pdf_subscription.id }
+    let(:user) { pdf_subscription.user }
+    let(:user_id) { user.id }
+
+    before do
+      allow(PdfSubscription).to receive(:find).with(pdf_subscription_id).and_return(pdf_subscription)
```

Comment:
This one is kind of tricky.  The find will work as expected, however, I don't think the subsequent stubbing of       `allow(user).to receive(:school_or_district_premium?).and_return(school_or_district_premium)` will work properly  since the `user` is something stored in memory from the let call rather than the `user` that from `PdfSubscription.find(pdf_subscription_id).user`.  There might be a better way to do this but this.

Reply:
Good point.  The only alternative I can think of to doing it this way is `allow_any_instance_of`, and we definitely don't want to do that.  This seems like a reasonable, if slightly awkward, solution to the problem and I don't have a better suggestion in mind.